packages/common/src/chains.ts

```diff
@@ -17,6 +17,24 @@ export enum ChainId {
   POLYGON_MUMBAI = 80001,
 }
 
+/**
+ * Attempts to parse a numerical or string chainId to the Enum.
+ * returns null if the chainid is invalid */
+export function tryParseChainIdToEnum(
+  chainId: string | number
+): ChainId | null {
+  const chains = Object.keys(ChainId)
+    .map(Number)
+    .filter((item) => {
+      return !isNaN(item);
+    });
```

Comment:
it would make sense to parse it at runtime if we have runtime values, but I see we use it with `allChains`, can't they be directly the right type so it's done at compile time?

Reply:
the id on wagmi's `Chain` type is number. this is not guaranteed to be a chain that we support. that's why we parse it.